monitoring/rid_qualifier/flight_track_factory.py

```diff
@@ -0,0 +1,178 @@
+            x0 =  self.minx + (u0 * cell_size_x)            
+            for v0 in range(0,2): # 2 rows
+                y0 = self.miny + (v0 *cell_size_y)
+                x1 = x0 + cell_size_x
+                y1 = y0 + cell_size_y
+                grid_cells.append(shapely.geometry.box(x0, y0, x1, y1))                
+        
+        self.flight_grid = grid_cells
+        
+
+    def generate_flight_paths_points(self):
+        ''' For each of the boxes allocated to the operator, get the centroid and buffer to get a flight path. A 75 m radius is provided to have flight paths within each of the boxes '''
+        # Iterate over the flight_grid
+        for grid_cell in self.flight_grid:
+            center = grid_cell.centroid
+            ## Transfrom to buffer 100 m diameter circle on which the drone will fly            
+            transformer = Transformer.from_crs("epsg:4326", "epsg:3857")
+            transformed_x,transformed_y = transformer.transform(center.x, center.y)
+            pt = Point(transformed_x,transformed_y)
+            buffer = pt.buffer(70) # build a buffer so that the radius is 70m for the track
+            buffer_points =  zip(buffer.exterior.coords.xy[0], buffer.exterior.coords.xy[1])
+            proj_buffer_points = []
+            # reproject back to ESPG 4326
+            transformer2 = Transformer.from_crs("epsg:3857", "epsg:4326")
+            for point in buffer_points:
+                x = point[0]
+                y = point[1]
+                x, y = transformer2.transform(x, y)
+                proj_buffer_points.append((x, y))
+            buffered_path = Polygon(proj_buffer_points)
+            altitude = 70.0
+            flight_points_with_altitude = []
+            x, y = buffered_path.exterior.coords.xy
+            for coord in range(0,len(x)):
+                flight_points_with_altitude.append((x[coord], y[coord], altitude))
+
+            # Build a list of points so that they can be fed to the sim and outputted. 
+            self.flight_points.append(flight_points_with_altitude)
+
+
+class TrackWriter():
+    ''' A class to write data geometry as files in geographical data format''' 
+
+    def __init__(self, path_points, bboxes) -> None:
+
+        self.flight_path_points = path_points
+        self.bboxes = bboxes
+        self.loop_counter = 0
+        
+        output_directory = Path('outputs').mkdir(parents=True, exist_ok=True) # Create outputs directory if it does not exist
+
+    def write_bboxes(self):
+        for buffered_bbox_details in self.bboxes:
+            
+            features = json.dumps({'type': 'Feature', 'properties': {}, 'geometry': shapely.geometry.mapping(buffered_bbox_details['shape'])})
+            with open('outputs/box_%s.geojson'% buffered_bbox_details['name'],'w') as f:
+                f.write(features)
+                
+    def write_tracks(self):
+        ''' This module writes tracks as a GeoJSON / KML for use in other software '''       
```

Comment:
Just GeoJSON (not KML), right?  And, that GeoJSON will also be ingested by aircraft_state_replayer, so it seems like it's not just for other software.  (Or, will we also write a proprietary format in addition to GeoJSON?  If so, that should probably go in this PR)

Reply:
Yes agreed, I think for the moment, GeoJSON should be fine, I am open to developing something else but I think we get this working and see if GeoJSON is OK. Eventually I envision the ability to pass on output format e.g. some people may prefer KML others GeoJSON , but that's for later.